src/main/java/org/elasticsearch/transport/RequestHandlerReg.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.transport;
+
+import org.elasticsearch.ElasticsearchIllegalStateException;
+
+import java.lang.reflect.Constructor;
+
+/**
+ *
+ */
+public class RequestHandlerReg<Request extends TransportRequest> {
+
+    private final String action;
+    private final Constructor<Request> request;
+    private final TransportRequestHandler<Request> handler;
+    private final boolean forceExecution;
+    private final String executor;
+
+    RequestHandlerReg(String action, Class<Request> request, TransportRequestHandler<Request> handler,
+                      String executor, boolean forceExecution) {
+        this.action = action;
+        try {
+            this.request = request.getDeclaredConstructor();
+        } catch (NoSuchMethodException e) {
+            throw new ElasticsearchIllegalStateException("failed to create constructor (does it have a default constructor?) for request " + request, e);
+        }
+        this.request.setAccessible(true);
```

Comment:
I'd like to create an instance of the Request in here just to make sure we fail early.